cunumeric/array.py

```diff
@@ -184,6 +186,140 @@ def _convert_all_to_numpy(obj: Any) -> Any:
         return obj
 
 
+def writeable() -> Callable[[Callable[P, R]], Callable[P, R]]:
+    """
+    Check if the current array is writeable
+    This decorator needs to be manually inserted
+    with consideration on the behavior of the corresponding method
+    """
+
+    def decorator(func: Callable[P, R]) -> Callable[P, R]:
+        def wrapper(*args: Any, **kwargs: Any) -> Any:
+            if isinstance(args[0], ndarray) and not args[0].flags.writeable:
+                raise RuntimeError("'self' is not writeable")
+            return func(*args, **kwargs)
+
+        return wrapper
+
+    return decorator
+
+
+class FlagsObj(object):
+    def __init__(
+        self,
+        c_contiguous: bool = True,
+        f_contiguous: bool = False,
+        owndata: bool = True,
+        writeable: bool = True,
+        aligned: bool = True,
+        writebackifcopy: bool = False,
+    ) -> None:
+        self._c_contiguous = c_contiguous
+        self._f_contiguous = f_contiguous
+        self._owndata = owndata
+        self._writeable = True
```

Comment:
should it be `self._writable = writable`? The same question for `-aligned` and `_writebackifcopy`

Reply:
Yes, I will fix the assignments.